exporter/splunkhecexporter/client.go

```diff
@@ -394,6 +394,14 @@ func (c *client) pushMetricsRecords(ctx context.Context, mds pmetric.ResourceMet
 
 		// Parsing metric record to Splunk event.
 		events := mapMetricToSplunkEvent(res.Resource(), metrics.At(k), c.config, c.logger)
+		if c.config.UseMultiMetricFormat {
+			merged, err := mergeEventsToMultiMetricFormat(events)
```

Comment:
Given that `events` list is made from just one OTLP metric, all the items will always have the same `metric_name:...` field key resulting into collisions. So it looks like `UseMultiMetricFormat` setting is useless in current implementation. Please correct me if I'm wrong.

Reply:
Some mappings such as histograms use a mapping that maps them to multiple metric names, and have distinct values. Right now each histogram value is sent separately for example.
We could run this for all metrics at once, but it's a bit more unlikely they have the same common fields such as timestamp.

Reply:
Can we do that for histograms only in that case? I believe histograms are not very popular (e.g. there is no scraper in the collector for histograms). Otherwise I think it's going to be a significant computational overhead which is no-op for most of the metrics. I would recommend moving this logic to `mapMetricToSplunkEvent` under histogram case and enabling it by default.

Reply:
Not all versions of Splunk support this format. So it has to be enabled.
It's not just histograms, others also create multiple metric values.

Reply:
I don't understand how other metric types can create multiple values if only one OTLP metric is passed to `mapMetricToSplunkEvent` with only one name -> the same `metric_name:...` HEC field -> collisions in `mergeEventsToMultiMetricFormat`. Please elaborate

Reply:
I see [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3dd3b94d228798385659ff8aec79e3ec62ffb642/exporter/splunkhecexporter/metricdata_to_splunk.go) that only summary and histograms can create multiple fields. Summary is even less popular than histogram and was added only for compatibility with other formats. I would recommend moving this logic to `mapMetricToSplunkEvent` to those to types of metrics. I believe this will significantly simplify this logic and remove computational overhead for other most common metric types.

Reply:
That's correct. However, I guess I have to think harder about this. This requirement was from the report, with this type of grouping:
```
{
  "time": 1486683865,
  "event": "metric",
  "source": "metrics",
  "sourcetype": "perflog",
  "host": "host_1.splunk.com",
  "fields": {
    "region": "us-west-1",
    "datacenter": "dc2",
    "rack": "63",
    "os": "Ubuntu16.10",
    "arch": "x64",
    "team": "LON",
    "service": "6",
    "service_version": "0",
    "service_environment": "test",
    "path": "/dev/sda1",
    "fstype": "ext3",
    "metric_name:cpu.usr": 11.12,
    "metric_name:cpu.sys": 12.23,
    "metric_name:cpu.idle": 13.34
  }
}
```
In this case, these are 3 separate gauge readings group together.
So maybe I got this all wrong, and I need to group not for one metric but for all metrics. That follows your feedback.